### PR TITLE
Task for generating unique build id

### LINKED ISSUE #368: Task for generating unique build id
Description
The purpose of this feature request is to create a task definition that is able to generate a unique build Id. A build Id is often appended to base version Ids in JAR files, Docker image tags to make them unique. If such as task is made available, then it would help a number of users, who currently use different CI systems (like Jenkins) that automatically provide a unique build Id to them, adopt Tekton pipelines in their CI/CD workflow.

### PR DESCRIPTION
**Pull request by maneeshmehra**
# Changes
Task for generating unique build id

Closes #368 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by chmouel** [comment]
/lgtm

Looking good thanks, documentation is top notch too so extra 👍  for that!

**Comment by maneeshmehra** [comment]
@chmouel: Thanks for the suggestion and reviewing my changes. Excited to have this PR merged, hopefully soon.

File: buildid/examples/taskrun.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
```
Comment (savitaashture) [inline buildid/examples/taskrun.yaml:2]: I believe  `TaskRun` has already been covered as part of `buildid/tests` 
IMO having it again in examples folder will be useful if we are showing something different scenario like `buildid/examples/pipeline.yaml`

Let me know what you think on this

Comment (maneeshmehra) [reply buildid/examples/taskrun.yaml:2]: I debated that too and decided that I would keep the content in the tests different than the one in examples as over time I might have more tests added to the same test configuration without necessarily updating the example (IMO the example should be fairly simple for anyone to understand)

**Comment by maneeshmehra** [comment]
@savitaashture, @vdemeester: Could you please tell me why the integration tests are failing by looking at the logs ? It seems to me that the failures are unrelated to my changes and has something to do with the kaniko tests, but I could be wrong.

From the [execution log](https://storage.googleapis.com/tekton-prow/pr-logs/pull/tektoncd_catalog/369/pull-tekton-catalog-integration-tests/1273713401905811456/build-log.txt) of the tests:

```
...
+ kubectl logs --all-containers -n kaniko-547 pod/affinity-assistant-74cd7a542a-0
Error from server (BadRequest): container "affinity-assistant" in pod "affinity-assistant-74cd7a542a-0" is terminated
+ teardown_test_resources
+ ((  IS_BOSKOS  ))
+ return
2020/06/18 20:39:03 process.go:155: Step '/home/prow/go/src/github.com/tektoncd/catalog/test/e2e-tests.sh --run-tests' finished in 9m30.826190893s
2020/06/18 20:39:03 main.go:319: Something went wrong: encountered 1 errors: [error during /home/prow/go/src/github.com/tektoncd/catalog/test/e2e-tests.sh --run-tests: exit status 1]
++ get_test_return_code
+++ cat /tmp/tcatalog-e2e-result
++ echo 1
...
```

**Comment by vdemeester** [comment]
@maneeshmehra yep, seems like a "github" glitch :sweat: 
/test pull-tekton-catalog-integration-tests

**Comment by maneeshmehra** [comment]
Thanks for re-running the tests. They passed this time around.

**Comment by maneeshmehra** [comment]
Could someone tell me why is there a "do-not-merge/invalid-owners-file" tag on this PR ?

**Comment by vdemeester** [comment]
@maneeshmehra  switched the base branch to master as it got merged with `v1beta1` now :angel: 
Concerning the `invalid-owners-file`, this means you need to be part of the organization to be able to be in an `OWNERs` file. See https://github.com/tektoncd/community/pull/116 as an example of PR to `tektoncd/community` to add yourself to the organization.

**Comment by maneeshmehra** [comment]
I have created: https://github.com/tektoncd/community/pull/132 to add myself as a new member of the org.

**Comment by maneeshmehra** [comment]
Reviewers: We have all the tests passing and https://github.com/tektoncd/community/pull/132 has been merged too now. Do you think you can remove the `do-not-merge/invalid-owners-file` tag and merge this PR now ? TIA.

**Comment by chmouel** [comment]
/verify-owners

Let's try if that does it ?

**Comment by chmouel** [comment]
@vdemeester @bobcatfish if someone can help us removing the invalid-owner-files tag...

**Comment by vdemeester** [comment]
@chmouel yeah for some reason our prow doesn't support the `/verify-owners` command…

**Comment by chmouel** [comment]
Thanks @vdemeester 

I believe we need to merge this first https://github.com/tektoncd/catalog/pull/383 to unblock the CI

**Comment by chmouel** [comment]
/lgtm

about time 😅

**Comment by maneeshmehra** [comment]
YIPPEE !!! Thanks a ton to @chmouel, @vdemeester and @savitaashture for reviewing my changes and supporting me each step of the way !

**Comment by chmouel** [comment]
@maneeshmehra Thanks for your contribution! 🙏

**Pull request merged**